src/packages/frontend/project/page/flyouts/store.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ *  This file is part of CoCalc: Copyright © 2024 Sagemath, Inc.
+ *  License: MS-RSL – see LICENSE.md for details
+ */
+
+import { merge, sortBy, throttle, uniq, xor } from "lodash";
+import { useState } from "react";
+import useAsyncEffect from "use-async-effect";
+
+import api from "@cocalc/frontend/client/api";
+import { STARRED_FILES } from "@cocalc/util/consts/bookmarks";
+import {
+  GetStarredBookmarks,
+  GetStarredBookmarksPayload,
+  SetStarredBookmarks,
+} from "@cocalc/util/types/bookmarks";
+import {
+  FlyoutActiveStarred,
+  getFlyoutActiveStarred,
+  storeFlyoutState,
+} from "./state";
+
+// Additionally to local storage, we back the state of the starred files in the database.
+// Errors with the API are ignored, because we primarily rely on local storage.
+// The only really important situation to think of are when there is nothing in local storage but in the database,
+// or when there is
```

Comment:
This comment ends midsentence: "The only really important situation to think of[...] or when there is [NOTHING]"